lib/vdsm/storage/mpathconf.py

```diff
@@ -130,3 +167,44 @@ def read_blacklist():
         wwids.add(wwid)
 
     return wwids
+
+
+def check_conf_revision():
+    """
+    The multipath configuration file at /etc/multipath.conf should contain
+    a tag in form "RHEV REVISION X.Y" for this check to succeed.
+    If the tag above is followed by tag "RHEV PRIVATE" the configuration
+    should be preserved at all cost.
+
+    Returns:
+        bool: check has passed.
+    """
+    first = second = ''
+    with open(CONF_FILE) as f:
+        mpathconf = [x.strip("\n") for x in f.readlines()]
+    try:
+        first = mpathconf[0]
+        second = mpathconf[1]
+    except IndexError:
+        pass
+    if _PRIVATE_TAG in second or _OLD_PRIVATE_TAG in second:
+        log.info("Manual override for multipath.conf detected"
+                 " - preserving current configuration\n")
```

Comment:
log.info does not work in the context of running vdsm tool. The tool logs to stdout now,
and we don't want to change this now. To move the code to storage, we need to change
the interface so this function does not return a boolean but a enum describing the state of the system.

I think in this case the function should return this info:
- revision: OK, OLD, MISSING
- private: bool

The caller will handle warn about the various states in the context of the configurator.

The lvmfilter module is a good example for this kind of code.